pkg/api/dirs.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// dirUploadInfo contains the data for a directory to be uploaded
+type dirUploadInfo struct {
+	toEncrypt bool
```

Comment:
this is kind of meh. i find it better to inject these parameters onto the context object itself, then just pass the context to the underlying components that do the actual work. then you can safely avoid using all of those `toEncrypt` fields all over the place

Reply:
yeah... you're right. i initially started out with this struct since it had more fields, and was mainly inspired by the `fileUploadInfo` struct. but i don't see much of a reason to keep it around at this point.

i've gone with the context route as you suggested. please have another look and let me know your thoughts on this current implementation.

thanks